File: packages/jsii-pacmak/test/expected.jsii-calc/dotnet/Amazon.JSII.Tests.CalculatorPackageId/Amazon/JSII/Tests/CalculatorNamespace/Constructors.cs
using Amazon.JSII.Runtime.Deputy;

namespace Amazon.JSII.Tests.CalculatorNamespace
{
    /// <remarks>stability: Experimental</remarks>
    [JsiiClass(nativeType: typeof(Constructors), fullyQualifiedName: "jsii-calc.Constructors")]
    public class Constructors : DeputyBase
    {
        public Constructors(): base(new DeputyProps(new object[]{}))
        {
        }

        protected Constructors(ByRefValue reference): base(reference)
        {
        }

        protected Constructors(DeputyProps props): base(props)
        {
        }

        /// <remarks>stability: Experimental</remarks>
        [JsiiMethod(name: "hiddenInterface", returnsJson: "{\"type\":{\"fqn\":\"jsii-calc.IPublicInterface\"}}")]
        public static IIPublicInterface HiddenInterface()
        {
            return InvokeStaticMethod<IIPublicInterface>(typeof(Constructors), new object[]{});
        }

        /// <remarks>stability: Experimental</remarks>
        [JsiiMethod(name: "hiddenInterfaces", returnsJson: "{\"type\":{\"collection\":{\"kind\":\"array\",\"elementtype\":{\"fqn\":\"jsii-calc.IPublicInterface\"}}}}")]
        public static IIPublicInterface[] HiddenInterfaces()
        {
            return InvokeStaticMethod<IIPublicInterface[]>(typeof(Constructors), new object[]{});
        }

        /// <remarks>stability: Experimental</remarks>
        [JsiiMethod(name: "hiddenSubInterfaces", returnsJson: "{\"type\":{\"collection\":{\"kind\":\"array\",\"elementtype\":{\"fqn\":\"jsii-calc.IPublicInterface\"}}}}")]
        public static IIPublicInterface[] HiddenSubInterfaces()
        {
            return InvokeStaticMethod<IIPublicInterface[]>(typeof(Constructors), new object[]{});
        }

        /// <remarks>stability: Experimental</remarks>
        [JsiiMethod(name: "makeClass", returnsJson: "{\"type\":{\"fqn\":\"jsii-calc.PublicClass\"}}")]
        public static PublicClass MakeClass()
        {
            return InvokeStaticMethod<PublicClass>(typeof(Constructors), new object[]{});
        }

        /// <remarks>stability: Experimental</remarks>
        [JsiiMethod(name: "makeInterface", returnsJson: "{\"type\":{\"fqn\":\"jsii-calc.IPublicInterface\"}}")]
        public static IIPublicInterface MakeInterface()
        {
            return InvokeStaticMethod<IIPublicInterface>(typeof(Constructors), new object[]{});
        }

        /// <remarks>stability: Experimental</remarks>
        [JsiiMethod(name: "makeInterface2", returnsJson: "{\"type\":{\"fqn\":\"jsii-calc.IPublicInterface2\"}}")]
        public static IIPublicInterface2 MakeInterface2()
        {
            return InvokeStaticMethod<IIPublicInterface2>(typeof(Constructors), new object[]{});
        }

        /// <remarks>stability: Experimental</remarks>
        [JsiiMethod(name: "makeInterfaces", returnsJson: "{\"type\":{\"collection\":{\"kind\":\"array\",\"elementtype\":{\"fqn\":\"jsii-calc.IPublicInterface\"}}}}")]
        public static IIPublicInterface[] MakeInterfaces()
        {
            return InvokeStaticMethod<IIPublicInterface[]>(typeof(Constructors), new object[]{});
        }
    }
}

File: packages/jsii-pacmak/test/expected.jsii-calc/kotlin/src/main/kotlin/software/amazon/jsii/tests/kotlin/calculator/Constructors.kt
package software.amazon.jsii.tests.kotlin.calculator

@javax.annotation.Generated("jsii-pacmak")
@software.amazon.jsii.Jsii(module = software.amazon.jsii.tests.kotlin.calculator.`$Module`::class, fqn = "jsii-calc.Constructors")
@software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
open class Constructors : software.amazon.jsii.JsiiObject {
    companion object {
        @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        fun hiddenInterface(): software.amazon.jsii.tests.kotlin.calculator.IPublicInterface {
            return software.amazon.jsii.JsiiObject.jsiiStaticCall(software.amazon.jsii.tests.kotlin.calculator.Constructors::class.java, "hiddenInterface", software.amazon.jsii.tests.kotlin.calculator.IPublicInterface::class.java) ?: error("Method 'hiddenInterface()' returned null value")
        }

        @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        fun hiddenInterfaces(): kotlin.collections.List<software.amazon.jsii.tests.kotlin.calculator.IPublicInterface> {
            return software.amazon.jsii.JsiiObject.jsiiStaticCall(software.amazon.jsii.tests.kotlin.calculator.Constructors::class.java, "hiddenInterfaces", kotlin.collections.List::class.java) as? kotlin.collections.List<software.amazon.jsii.tests.kotlin.calculator.IPublicInterface> ?: error("Method 'hiddenInterfaces()' returned null value")
        }

        @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        fun hiddenSubInterfaces(): kotlin.collections.List<software.amazon.jsii.tests.kotlin.calculator.IPublicInterface> {
            return software.amazon.jsii.JsiiObject.jsiiStaticCall(software.amazon.jsii.tests.kotlin.calculator.Constructors::class.java, "hiddenSubInterfaces", kotlin.collections.List::class.java) as? kotlin.collections.List<software.amazon.jsii.tests.kotlin.calculator.IPublicInterface> ?: error("Method 'hiddenSubInterfaces()' returned null value")
        }

        @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        fun makeClass(): software.amazon.jsii.tests.kotlin.calculator.PublicClass {
            return software.amazon.jsii.JsiiObject.jsiiStaticCall(software.amazon.jsii.tests.kotlin.calculator.Constructors::class.java, "makeClass", software.amazon.jsii.tests.kotlin.calculator.PublicClass::class.java) ?: error("Method 'makeClass()' returned null value")
        }

        @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        fun makeInterface(): software.amazon.jsii.tests.kotlin.calculator.IPublicInterface {
            return software.amazon.jsii.JsiiObject.jsiiStaticCall(software.amazon.jsii.tests.kotlin.calculator.Constructors::class.java, "makeInterface", software.amazon.jsii.tests.kotlin.calculator.IPublicInterface::class.java) ?: error("Method 'makeInterface()' returned null value")
        }

        @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        fun makeInterface2(): software.amazon.jsii.tests.kotlin.calculator.IPublicInterface2 {
            return software.amazon.jsii.JsiiObject.jsiiStaticCall(software.amazon.jsii.tests.kotlin.calculator.Constructors::class.java, "makeInterface2", software.amazon.jsii.tests.kotlin.calculator.IPublicInterface2::class.java) ?: error("Method 'makeInterface2()' returned null value")
        }

        @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        fun makeInterfaces(): kotlin.collections.List<software.amazon.jsii.tests.kotlin.calculator.IPublicInterface> {
            return software.amazon.jsii.JsiiObject.jsiiStaticCall(software.amazon.jsii.tests.kotlin.calculator.Constructors::class.java, "makeInterfaces", kotlin.collections.List::class.java) as? kotlin.collections.List<software.amazon.jsii.tests.kotlin.calculator.IPublicInterface> ?: error("Method 'makeInterfaces()' returned null value")
        }
    }
    constructor(objRef: software.amazon.jsii.JsiiObjectRef) : super(objRef)
    constructor(mode: software.amazon.jsii.JsiiObject.InitializationMode) : super(mode)

    constructor() : super(software.amazon.jsii.JsiiObject.InitializationMode.JSII) {
        software.amazon.jsii.JsiiEngine.getInstance().createNewObject(this)
    }
}
